server/subsonic/middlewares.go

```diff
@@ -148,7 +148,7 @@ func getPlayer(players core.Players) func(next http.Handler) http.Handler {
 			userName, _ := request.UsernameFrom(ctx)
 			client, _ := request.ClientFrom(ctx)
 			playerId := playerIDFromCookie(r, userName)
-			ip, _, _ := net.SplitHostPort(realIP(r))
+			ip, _, _ := net.SplitHostPort(r.RemoteAddr)
```

Comment:
We may have discussed this before, but I don't see any reason to change this. This could cause multiple clients behind a cache proxy to use the same RemoteAddr.

Reply:
The call to `realIP` has not been removed, but moved out of the `getPlayer` function and just before in the middleware chain (the other line you highlighted in `server/subsonic/api.go`). At this point, because the `realIP` middleware has already been applied, `r.RemoteAddr` contains the substituted real IP.